src/mumble/ServerHandler.cpp

```diff
@@ -392,6 +393,11 @@ void ServerHandler::sendPing() {
 	if (!connection)
 		return;
 
+	if (iInFlightTCPPings >= g.s.iMaxInFlightTCPPings) {
+		serverConnectionClosed(QAbstractSocket::UnknownSocketError, tr("Server is not responding to TCP pings"));
```

Comment:
> [...] and custom even exit logic.
For the one case it is used (connected to server has no certificate) we don’t have to close Mumble, but just the Connection.

Hmm? This isn't ::exit(0), it's QThread::exit(), which makes the ServerHandler thread terminate.